src/devices/transitions/transitionHandler.ts

```diff
@@ -0,0 +1,178 @@
+import { TSRTransitionOptions } from '../../types/src/casparcg'
+import * as _ from 'underscore'
+import { Animator, LinearMovement, PhysicalAcceleration } from './animate'
+
+
+
+interface TransitionHandler {
+	values: number[]
+	target: number[]
+	groups: string[]
+
+	calculatingGroups: {[groupId: string]: {
+		animator: Animator
+	}}
+
+	activeIterator: NodeJS.Timeout | null
+	lastUpdate: number
+	updateCallback?: (newValues: number[]) => void
+}
+export class InternalTransitionHandler {
+	private _transitions: {[identifier: string]: TransitionHandler} = {}
+
+	public terminate () {
+		// clearInterval(this._interval)
+		_.each(this._transitions, (_transition, identifier) => {
+			this.clearTransition(identifier)
+		})
+	}
+	public getIdentifiers () {
+		return Object.keys(this._transitions)
+	}
+
+	public clearTransition (identifier: string) {
+		const t = this._transitions[identifier]
+		if (t) {
+			this._stopTransition(t)
+			delete this._transitions[identifier]
+		}
+	}
+	public snapTransition (
+		identifier: string,
+		targetValues: number[]
+	) {
+		if (!this._transitions[identifier]) {
+			this.initTransition(identifier, targetValues)
+		}
+		const t = this._transitions[identifier]
+
+		this._stopTransition(t)
+
+		t.values = targetValues
+	}
+	private initTransition (
+		identifier: string,
+		initialValues: number[]
+	) {
+		// Set initial values:
+		this._transitions[identifier] = {
+
+			values: initialValues,
+			target: [], // filled in later
+			groups: [], // filled in later
+
+			activeIterator: null,
+			lastUpdate: 0,
+
+			calculatingGroups: {}
+		}
+	}
+	public activateTransition (
+		identifier: string,
+		initialValues: number[],
+		targetValues: number[],
+		groups: string[],
+		options: TSRTransitionOptions,
+		animatorTypes: {[groupId: string]:
+			{
+				type: 'linear' | 'physical',
+				options?: TSRTransitionOptions
+			}
+		},
+		updateCallback: (newValues: number[]) => void
+	) {
+		// Note: this is a preliminary implemenation, that animates x & y, and snaps xScale & yScale
```

Comment:
Which bit is preliminary? It seems to iterate generically. But if this is still true it would be better to put it into a jsdoc style comment so the editor will show it when using the method.

Reply:
It was preliminary at the time of writing, should be definite by now 👍